test/bench/bench.cpp

```diff
@@ -59,54 +73,65 @@ struct ExecutionBenchmarkCase
     fizzy::bytes expected_memory;
 };
 
-void benchmark_execute(benchmark::State& state, const ExecutionBenchmarkCase& benchmark_case)
+void benchmark_execute(
+    benchmark::State& state, EngineCreateFn create_fn, const ExecutionBenchmarkCase& benchmark_case)
 {
-    const auto module = fizzy::parse(*benchmark_case.wasm_binary);
-    const auto func_idx = fizzy::find_exported_function(module, benchmark_case.func_name);
+    const auto engine = create_fn();
+    engine->parse(*benchmark_case.wasm_binary);
+    const auto func_idx = engine->find_function(benchmark_case.func_name);
     if (!func_idx)
     {
         return state.SkipWithError(
             ("Function \"" + benchmark_case.func_name + "\" not found").c_str());
     }
 
-    auto instance = fizzy::instantiate(module, {});
+    engine->instantiate();
+
+    auto initial_memory = fizzy::bytes{engine->get_memory()};
 
     // TODO: Check if memory is exported or imported.
-    if (benchmark_case.memory.size() > instance.memory.size())
+    if (benchmark_case.memory.size() > initial_memory.size())
         return state.SkipWithError("Cannot init memory");
+
     std::copy(std::begin(benchmark_case.memory), std::end(benchmark_case.memory),
-        std::begin(instance.memory));
-    const auto initial_memory = instance.memory;
+        std::begin(initial_memory));
+    engine->set_memory(initial_memory);
 
     {  // Execute once and check results against expectations.
-        const auto result = fizzy::execute(instance, *func_idx, benchmark_case.func_args);
+        const auto result = engine->execute(*func_idx, benchmark_case.func_args);
         if (result.trapped)
             return state.SkipWithError("Trapped");
 
         if (benchmark_case.expected_result)
         {
-            if (result.stack.size() != 1)
-                return state.SkipWithError("Incorrect number of results");
-            else if (result.stack[0] != *benchmark_case.expected_result)
+            if (!result.value)
+                return state.SkipWithError("Missing result value");
+            else if (*result.value != *benchmark_case.expected_result)
                 return state.SkipWithError("Incorrect result");
         }
-        else if (!result.stack.empty())
+        else if (result.value)
             return state.SkipWithError("Unexpected result");
 
-        if (instance.memory.size() < benchmark_case.expected_memory.size())
+        const auto mem = engine->get_memory();
```

Comment:
Final nit: perhaps use `memory` (as we use long form mostly in the codebase). But I don't care.